Tests/ApolloCodegenTests/CodeGeneration/Templates/SelectionSet/SelectionSetTemplateTests.swift

```diff
@@ -1488,787 +1488,771 @@ class SelectionSetTemplateTests: XCTestCase {
+      id: String
+      species: String
+      genus: String
+      name: String
+      friend: Animal
+    }
+
+    type Cat implements Animal {
+      id: String
+      species: String
+      genus: String
     }
     """
 
     document = """
-    query TestOperation($a: Boolean!) {
+    query TestOperation {
       allAnimals {
-        ...FragmentA @include(if: $a)
+        __typename
+        id
+        ... on Dog @defer(label: "one") {
+          friend {
+            ... on Cat @defer(label: "two") {
+              species
+            }
+          }
+        }
       }
     }
-
-    fragment FragmentA on Pet {
-      int
-    }
     """
 
-    let expected = """
+    let expectedOne = """
       public static var __selections: [ApolloAPI.Selection] { [
-        .field("__typename", String.self),
-        .include(if: "a", .inlineFragment(AsPetIfA.self)),
+        .deferred(One.self, label: "one"),
+      ] }
+    """
+    let expectedTwo = """
+      public static var __selections: [ApolloAPI.Selection] { [
+        .deferred(Two.self, label: "two"),
       ] }
     """
 
     // when
     try buildSubjectAndOperation()
-    let allAnimals = try XCTUnwrap(
-      operation[field: "query"]?[field: "allAnimals"] as? IR.EntityField
+    let allAnimals_asDog = try XCTUnwrap(
+      operation[field: "query"]?[field: "allAnimals"]?[as: "Dog"]
+    )
+    let allAnimals_asDog_asCat = try XCTUnwrap(
+      allAnimals_asDog[as: "One"]?[field: "friend"]?[as: "Cat"]
     )
 
-    let actual = subject.render(field: allAnimals)
+    let actualOne = subject.render(inlineFragment: allAnimals_asDog)
+    let actualTwo = subject.render(inlineFragment: allAnimals_asDog_asCat)
 
     // then
-    expect(actual).to(equalLineByLine(expected, atLine: 7, ignoringExtraLines: true))
+    expect(actualOne).to(equalLineByLine(expectedOne, atLine: 11, ignoringExtraLines: true))
+    expect(actualTwo).to(equalLineByLine(expectedTwo, atLine: 11, ignoringExtraLines: true))
   }
 
-  func test__render_selections__givenInlineFragmentOnSameTypeWithConditions_rendersInlineFragmentSelectionSetAccessorWithCorrectName() throws {
+  // MARK: Selections - Deferred Named Fragment
+
+  func test__render_selections__givenDeferredNamedFragment_rendersDeferredFragmentSelection() throws {
+    throw XCTSkip()
+
     // given
     schemaSDL = """
     type Query {
       allAnimals: [Animal!]
     }
 
-    type Animal {
-      fieldA: String!
+    interface Animal {
+      id: String
+      species: String
+      genus: String
     }
 
-    interface Pet {
-      fieldA: String!
+    type Dog implements Animal {
+      id: String
+      species: String
+      genus: String
+      name: String
     }
     """
 
     document = """
-    query TestOperation($a: Boolean!) {
+    query TestOperation {
       allAnimals {
-        ... on Animal @include(if: $a) {
-          fieldA
-        }
+        __typename
+        id
+        ...DogFragment @defer(label: "root")
```

Comment:
Will be handled in a subsequent PR that focuses on the selection set generation.

Reply:
> I can't remember if we came to the decision that we don't need the label argument for deferring named fragments? It looks like in the DeferCondition struct we have label as required. Was there a reason why we decided we still needed the label? Was it for execution or something?

I remember that conversation and yes I think we came to the conclusion that it may not be necessary because we already have something to name it in the named fragment. I would prefer to keep it around for now incase, as you mentioned, it's needed for execution. I created https://github.com/apollographql/apollo-ios/issues/3256 so we can make that change later if we find the label is not needed anywhere.